RFunction.R

```diff
@@ -436,31 +416,67 @@ rFunction = function(data, travelcut,
   )
   
   # Log results
-  logger.trace(paste0("   ", sum(data$behav == "SResting", na.rm = T), " locations classified as SResting"))
-  logger.trace(paste0("   ", sum(data$behav == "STravelling", na.rm = T), " locations classified as STravelling"))
+  logger.info(paste0("   |> ", sum(data$behav == "SResting", na.rm = T), " locations classified as SResting"))
+  logger.info(paste0("   |> ", sum(data$behav == "STravelling", na.rm = T), " locations classified as STravelling"))
   
   
-  #### 2. Altitude Classification ----
-  logger.info("[2] Performing altitude classification")
-  data %<>%
-    mutate(
-      RULE = case_when(
-        (behav == "SResting") & (altchange == "ascent") ~ "[2] Altitude increasing",
-        (behav == "SResting") & (altchange == "descent") & (lead(altchange) %in% c("descent", "ascent")) ~ "[2] Altitude decreasing",
-        TRUE ~ RULE
-      ),
-      behav = case_when(
-        (behav == "SResting") & (altchange == "ascent") ~ "STravelling",
-        (behav == "SResting") & (altchange == "descent") & (lead(altchange) %in% c("descent", "ascent")) ~ "STravelling",
-        TRUE ~ behav
+  ## [2] Altitude Classification ----
+  
+  #' Remaining resting locations reclassified as travelling according to the following rules:
+  #' (i) If a bird is ascending ==> STravelling
+  #' (ii) If a bird is descending AND:
+  #'      Next location is ascending/descending ==> STravelling
+  #'      Next location is flatlining ==> remains SResting
+  #' (iii) If a bird is flatlining, it remains SResting
+  #' 
+  #' NOTE: stationary locations are not re-classified here, so this step can
+  #' produce locations with both `stationary == 1` AND `behav == STravelling`.
+  #' This can be a bit counter-intuitive, but we don't want to change it,
+  #' otherwise the roosting-site identification performed in Data Prep would be
+  #' outdated
+  
+  if(alt_classify){
+  
+    logger.info("[2] Performing altitude classification")
+    
+    data %<>%
+      # QUESTION (BC): shouldn't this step be grouped by bird given we're using `lead()`?
+      # group_by(ID) %>%
+      mutate(
+        RULE = case_when(
+          (behav == "SResting") & (altchange == "ascent") ~ "[2] Altitude increasing",
+          (behav == "SResting") & (altchange == "descent") & (lead(altchange) %in% c("descent", "ascent")) ~ "[2] Altitude decreasing",
+          TRUE ~ RULE
+        ),
+        behav = case_when(
+          (behav == "SResting") & (altchange == "ascent") ~ "STravelling",
+          (behav == "SResting") & (altchange == "descent") & (lead(altchange) %in% c("descent", "ascent")) ~ "STravelling",
+          TRUE ~ behav
+        ),
+        # # QUESTION (BC): Change stationary state to conform with above resting -> travelling re-classification??
```

Comment:
maybe makes sense.  depends on our definition of stationary.  if its just speed < travel cut then maybe we dont change.

Reply:
Expanded in issue #11. Indeed, it boils down to the definition of stationary - are we good with the status of traveling while stationary? 😄 I guess the interesting bit is the increase in # feeding events when this update is implemented, as described in #11.